src/interval.jl

```diff
@@ -233,20 +233,32 @@ function Base.maximum(interval::AbstractInterval{T,L,Open}; increment=nothing) w
     throw(BoundsError(interval, next_val))
 end
 
-##### CONVERSION #####
+"""
+    only(interval::AbstractInterval)
 
-# Allows an interval to be converted to a scalar when the set contained by the interval only
-# contains a single element.
-function Base.convert(::Type{T}, interval::Interval{T}) where T
+Returns the only element of a closed interval with coinciding endpoints (throws a
+DomainError otherwise).
+"""
+function Base.only(interval::Interval)
     if first(interval) == last(interval) && isclosed(interval)
         return first(interval)
     else
         throw(DomainError(interval, "The interval is not closed with coinciding endpoints"))
     end
 end
 
-##### DISPLAY #####
+##### CONVERSION #####
 
+function Base.convert(::Type{T}, interval::Interval{T}) where {T}
+    depwarn(
+        "`convert(::Type{T}, interval::Interval{T})` is deprecated, " *
+        "use `only(interval)` instead.",
+        :convert,
+    )
+    return only(interval)
+end
```

Comment:
Move this to `src/deprecated.jl`. You may be able to define this as:
```julia
@deprecated (convert(::Type{T}, interval::Interval{T}) where T) only(interval)
```
Also, take note there are some deprecations near the top that need to be updated to use `only`

Reply:
I don't think this will work, if we do, it creates a `Intervals.convert` that overshadows `Base.convert`, and we cannot use `Base.convert` within `@deprecated`.